woocommerce/grow
# Upgrade the npm dependency `jsdoc` to v4 for the JS package `jsdoc`

### Changes proposed in this Pull Request:

Please also review PR #103 and approve it as well once this PR is getting approval.

This PR upgrades the npm dependency `jsdoc` to v4 for the JS package `jsdoc` along with related deps

#### 📌 Checklist before merging (@eason9487)

- [x] Wait for #103 getting merged
- [x] Change to use the above merged revision
      https://github.com/woocommerce/grow/blob/e586897d7fcceaf44fd5fcd137ac3af2b554d09e/packages/js/jsdoc/package.json#L27

### Detailed test instructions:

#### Check the validation sample for the `jsdoc` and `tracking-jsdoc` JS packages with npm `jsdoc` v3

1. Set `node` to use v14
2. `git checkout 1485e2d9c3cf5eeb0c80815ec8564cf954470cf0` (Ref: 1485e2d9c3cf5eeb0c80815ec8564cf954470cf0)
3. `cd packages/js/jsdoc`
4. `npm install`
5. `npm ls jsdoc` to check if the current `jsdoc` is v3
6. `npm run dev:doc:tracking`
7. View the generated content in `samples/jsdoc/TRACKING.md` before the upgrade

#### Test the changes with `jsdoc` v4

1. Set `node` to use v14
2. `git checkout update/jsdoc-deps`
3. `cd packages/js/jsdoc`
4. `npm install`
5. `npm ls jsdoc` to check if the current `jsdoc` is v4
6. `npm run dev:doc:tracking`
7. Check if the generated content in `samples/jsdoc/TRACKING.md` is the same after the upgrade
8. Inspect the generated content in a package user side [Google Listings & Ads](https://github.com/woocommerce/google-listings-and-ads/blob/103a4f4f7f7a1ab084a1b33bec879ed680db5ef2/src/Tracking/README.md)
